src/condor_utils/submit_utils.h

```diff
@@ -642,6 +644,10 @@ class SubmitHash {
 	bool AssignJobVal(const char * attr, long val) { return AssignJobVal(attr, (long long)val); }
 	//bool AssignJobVal(const char * attr, time_t val)  { return AssignJobVal(attr, (long long)val); }
 
+	//Set job submit method to enum equal to passed value if value is in range
+	void setSubmitMethod(int value){ s_method = (value < -1 || value > 6) ? submit_method::UNDEFINED : static_cast<submit_method>(value); }
```

Comment:
remove the range check here,  either allow any value or change the type to the enum type instead of int